orangecontrib/storynavigation/modules/tagging.py

```diff
@@ -326,43 +326,118 @@ def __process_dutch_future_verbs(self, tags, tag):
 
         return "-"  # if no future tense verb is detected
     
+    
     def __process_dutch_voice(self, tags):
         """
-        Determine if a Dutch sentence or clause is in the passive or active voice.
+        Determine if a Dutch sentence is in the passive or active voice.
 
         Args:
             tags (list): A list of tags representing tokens in the sentence or clause.
 
         Returns:
             str: "PASSIVE" if the sentence/clause is in the passive voice, otherwise "ACTIVE".
         """
-        # Auxiliary verb indicating passive voice
-        passive_auxiliary_lemma = "worden"
-        
-        # Track whether "worden" and a past participle are found
-        passive_triggered = False
-
-        # Loop through tokens in the sentence
+        # Lemmas of auxiliary verbs indicating passive voice
+        #passive_auxiliaries = {"worden", "zijn"}
+        passive_auxiliaries = {"worden"}
+        past_participle_tag_prefix = "WW|vd"  # past participles
+        auxiliary_pos_tag = "AUX"
+
+        # Flags to track detection of relevant elements
+        auxiliary_found = False
+        passive_auxiliary_found = False
+        past_participle_found = False
+        prepositional_agent_found = False  # Tracks presence of a "door" or similar preposition
+
+        # Keep track of auxiliary heads for context
+        auxiliary_heads = set()
+
+        # Iterate through the tags
         for tok in tags:
             lemma_value = tok[5].lemma_
+            pos_value = tok[1]  
+            tag_value = tok[2]  
+            dep_value = tok[3]  # Dependency relation
+            head_value = tok[5].head.text  # Syntactic head (index of the parent token)
```

Comment:
```suggestion
            head_value = tok[5].head.lemma_
```
Use "lemma_" here, like in the rest of the code